packages/engine-server/src/store/NoteMetadataStore.ts

```diff
@@ -118,4 +140,40 @@ export class NoteMetadataStore implements IDataStore<string, NotePropsMeta> {
 
     return { data: key };
   }
+
+  /**
+   * Recursively search through fname to find next available ancestor note.
+   *
+   * E.g, if fpath = "baz.foo.bar", search for "baz.foo", then "baz", then "root" until first valid note is found
+   * @param fpath of note to find ancestor of
+   * @param vault of ancestor note
+   * @returns closest ancestor note
+   */
+  private async findClosestAncestor(
```

Comment:
why would we do this and not use parent pointers in the NoteProps?

Reply:
This is for when we are creating a brand new note and we don't know know its parent yet